src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -363,50 +355,68 @@ class npc_snobold_vassal : public CreatureScript
                 switch (action)
                 {
                     case ACTION_ENABLE_FIRE_BOMB:
-                        _events.ScheduleEvent(EVENT_FIRE_BOMB, urand(5*IN_MILLISECONDS, 30*IN_MILLISECONDS));
+                        _events.ScheduleEvent(EVENT_FIRE_BOMB, randtime(Seconds(5), Seconds(30)));
                         break;
                     case ACTION_DISABLE_FIRE_BOMB:
                         _events.CancelEvent(EVENT_FIRE_BOMB);
                         break;
+                    case ACTION_ACTIVE_SNOBOLD:
+                        _isActive = true;
+                        break;
                     default:
                         break;
                 }
             }
 
-            void UpdateAI(uint32 diff) override
+            void SetGUID(ObjectGuid guid, int32 id) override
             {
-                if (!UpdateVictim() || _targetDied)
+                if (id == DATA_NEW_TARGET)
+                    if (Unit* target = ObjectAccessor::GetPlayer(*me, guid))
+                    {
+                        _targetGUID = guid;
+                        AttackStart(target);
+                        _events.ScheduleEvent(EVENT_BATTER, Seconds(5));
+                        _events.ScheduleEvent(EVENT_HEAD_CRACK, Seconds(25));
+                        _events.ScheduleEvent(EVENT_SNOBOLLED, Milliseconds(500));
+                    }
+            }
+
+            void AttackStart(Unit* who) override
+            {
+                //Snobold only melee attack players that is your vehicle
+                if (!_isActive || who->GetGUID() != _targetGUID)
                     return;
 
-                if (Unit* target = ObjectAccessor::GetPlayer(*me, _targetGUID))
+                ScriptedAI::AttackStart(who);
```

Comment:
Might want to disable attack move somewhere, just in case? They should never chase, after all.

Not sure if there's ever a situation where they're on the ground and _could_ chase, mind you.